File: Array/09.FindAllDuplicates/FindAllDuplicates.js
const findDuplicates = (nums) => {
    const res = [];
    for (let i = 0; i < nums.length; ++i) {
        const index = Math.abs(nums[i]) - 1;
        if (nums[index] < 0) {
            res.push(Math.abs(index) + 1);
        } else {
            nums[index] = -nums[index]
        }
    }
    return res;
}

const r = findDuplicates([1,2,3,2,7,5,6,1]);
console.log(r);

File: Array/09.FindAllDuplicates/FindAllDuplicates.cpp

class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> res;
        for(int i=0; i<nums.size(); ++i) {
            int index = abs(nums[i]) - 1;
            if(nums[index] < 0) { res.push_back(abs(nums[i])); }
            else { nums[index] = -nums[index]; }
        }
        return res;
    }
};